### PR TITLE
Add fused_inplace_qknorm_rope to HunyuanVideo model

### PR DESCRIPTION
**Pull request by pengdurice**
## Motivation

The `fused_inplace_qknorm_rope` JIT kernel ([#19059](https://github.com/sgl-project/sglang/pull/19059), [#21440](https://github.com/sgl-project/sglang/pull/21440), [#21503](https://github.com/sgl-project/sglang/pull/21503), [#21654](https://github.com/sgl-project/sglang/pull/21654)) and its `apply_qk_norm_with_optional_rope` helper have been the default code path for FLUX, FLUX.2, Qwen-Image, Z-Image, and Joy-Image since their introduction, but `runtime/models/dits/hunyuanvideo.py` still routes both `MMDoubleStreamBlock` and `MMSingleStreamBlock` through a hand-rolled `2× RMSNorm + 2× _apply_rotary_emb` chain at every site. This PR plumbs the existing helper into the four eligible call sites in HunyuanVideo, picking up a **~1.48× kernel speedup** on H200 (1.46× short clip / 1.50× medium clip) without changing model outputs.

This also addresses the "kernel fusion instead of compiler-driven fusion" line under Performance Improvements in [Roadmap] SGLang-Diffusion (26 Q2) (#23035).

## Modifications

- **`python/sglang/multimodal_gen/runtime/models/dits/hunyuanvideo.py`** — route four `RMSNorm + _apply_rotary_emb` sites through `apply_qk_norm_with_optional_rope`:
  - `MMDoubleStreamBlock.forward` — img-stream fused `QK-Norm + RoPE` (`cos_sin_cache=packed`) and txt-stream fused `QK-Norm only` (`cos_sin_cache=None`).
  - `MMSingleStreamBlock.forward` — image / text split is moved *before* QK-Norm so the image portion can route through fused `QK-Norm + RoPE` in a single in-place pass and the text portion through fused `QK-Norm` only. RMSNorm is a per-token operation, so reordering split vs. norm is mathematically a no-op (pinned by `test_single_stream_split_then_fuse_parity`).

  Drops the unused `_apply_rotary_emb` import; all four old `_apply_rotary_emb(img_q/img_k, cos, sin, is_neox_style=False)` sites are gone. `cos_sin_cache` is built inline at each block in the standard `torch.cat([cos.float().contiguous(), sin.float().contiguous()], dim=-1)` layout that FLUX / Wan already use just before the fused helper.

- **`python/sglang/jit_kernel/benchmark/diffusion/bench_qknorm_rope.py`** — add two HunyuanVideo shape cases (`hunyuanvideo_short` 1×8192×24×128, `hunyuanvideo_medium` 1×24576×24×128, both `rope_dim=128`) to the existing `BENCH_CASES` tuple. `num_heads=24`, `attention_head_dim=128`, `rope_axes_dim=(16, 56, 56)` per `configs/models/dits/hunyuanvideo.py`. The text-stream (~256 tokens, no RoPE) is covered by `bench_qknorm.py`.

- **`python/sglang/multimodal_gen/test/unit/test_hunyuanvideo_qknorm_rope_fusion.py`** (new file) — 9 CPU-runnable unit tests + 1 CUDA-gated bf16 test, covering pre-/post-refactor parity on img-stream norm+rope, txt-stream norm-only, single-stream split-first reordering, helper no-rope path, dtype preservation, and module wiring.

## Accuracy Tests

Zero-tolerance parity vs the original eager `RMSNorm + _apply_rotary_emb` chain on CPU, across 9 unit tests:

```bash
$ PYTHONPATH=python python -m pytest -p no:cacheprovider \
    python/sglang/multimodal_gen/test/unit/test_hunyuanvideo_qknorm_rope_fusion.py -v
...
============================ 9 passed in ~30s ============================
```

**Coverage:**

- `test_double_stream_img_norm_rope_parity` (2 parametrized cases) — img-stream pre-refactor `2× RMSNorm + 2× _apply_rotary_emb` vs fused helper, `torch.testing.assert_close(rtol=1e-5, atol=1e-5)`.
- `test_double_stream_txt_norm_only_parity` (2 parametrized cases) — txt-stream `2× RMSNorm` vs fused helper with `cos_sin_cache=None`.
- `test_single_stream_split_then_fuse_parity` (2 parametrized cases) — split-first pipeline matches full-norm-then-split-then-rope numerically; pins the reordering invariant that the single-stream refactor depends on.
- `test_helper_no_rope_path_matches_apply_qk_norm` — `cos_sin_cache=None` routes through plain QK-Norm.
- `test_helper_preserves_input_dtype_fp32` — fp32 round-trip without silent upcast (bf16 variant is CUDA-gated, see below).
- `test_hunyuanvideo_module_wires_fused_helper` — module-level wiring check: `apply_qk_norm_with_optional_rope` is bound and `_apply_rotary_emb` is no longer imported.

A CUDA-gated `test_helper_preserves_input_dtype_bf16_cuda` exists for bf16 dtype-preservation but is skipped on CPU-only hosts (sglang's `RMSNorm.forward_cuda` routes bf16/fp16 through `triton_one_pass_rms_norm`, which is registered for the CUDA backend only). The CUDA fast path of the fused helper itself is already covered by `python/sglang/jit_kernel/tests/diffusion/test_qknorm_rope.py`.

## Speed Tests and Profiling

Measured on H200 via the existing `bench_qknorm_rope.py` with the two HunyuanVideo cases added by this PR:

| case | shape | dtype | split (µs) | fused (µs) | speedup |
|---|---|---|---:|---:|---:|
| `flux_1024`         | 1×4096×24×128, rope_dim=128 | bf16 |  59.04 |  43.26 | 1.36× |
| `qwen_image_1024`   | 1×4096×32×128, rope_dim=128 | bf16 |  80.64 |  54.75 | 1.47× |
| `qwen_image_partial`| 1×4096×32×128, rope_dim=64  | bf16 |  79.14 |  54.43 | 1.45× |
| `zimage_1024`       | 1×4096×30×128, rope_dim=128 | bf16 |  74.02 |  50.72 | 1.46× |
| `batch2_medium`     | 2×2048×24×128, rope_dim=128 | bf16 |  59.30 |  43.10 | 1.38× |
| **`hunyuanvideo_short`**  | 1×8192×24×128, rope_dim=128  | bf16 | 115.46 |  78.88 | **1.46×** |
| **`hunyuanvideo_medium`** | 1×24576×24×128, rope_dim=128 | bf16 | 332.51 | 221.18 | **1.50×** |

Geomean kernel speedup on HunyuanVideo shapes: **1.48×**. The fused kernel is memory-bandwidth-bound on the read-norm-write-rope traffic — split path materializes the post-norm tensor to HBM before re-reading it for RoPE; fused path keeps it in registers across the RoPE pass — so the speedup scales mildly with token count.

Bench command:

```bash
PYTHONPATH=python CUDA_VISIBLE_DEVICES=0 python \
    python/sglang/jit_kernel/benchmark/diffusion/bench_qknorm_rope.py
```
## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

---
### CI States

Latest PR Test (Base): :x: [Run #27066413371](https://github.com/sgl-project/sglang/actions/runs/27066413371)
Latest PR Test (Extra): :x: [Run #27066413285](https://github.com/sgl-project/sglang/actions/runs/27066413285)

### CODE REVIEW

**Comment by pengdurice** [comment]
@mickqian Hi, would you mind review this PR? it is to add fused_inplace_qknorm_rope to Hunyuan video. Thank you!
